### PR TITLE
fixed bug for issue #1336

### PR DESCRIPTION
**Pull request by SadiHassan**
### Changelog category (leave one):
- Bug Fix (user-visible misbehavior in official stable or prestable release)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
...
Current input output:
```
print substring("123456", -2, 2)
┌─substring('123456', -2, 2)─┐
│ 56                         │
└────────────────────────────┘

print substring("ABCD", -2, 2)
┌─substring('ABCD', -2, 2)─┐
│ CD                       │
└──────────────────────────┘
```

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

File: src/Parsers/Kusto/KustoFunctions/KQLStringFunctions.cpp [hunk 1]
```diff
@@ -534,7 +534,13 @@ bool SubString::convertImpl(String & out,IParser::Pos & pos)
     {
         ++pos;
         auto length = getConvertedArgument(fn_name, pos);
-        out = "substr("+ source + ", " + startingIndex + " + 1, " + length + ")";
+
+        if(startingIndex.empty())
+            return false;
+        else if(startingIndex[0] == '-')
```
Comment (kashwy) [inline src/Parsers/Kusto/KustoFunctions/KQLStringFunctions.cpp:540]: do not check this in c++.

Comment (SadiHassan) [reply src/Parsers/Kusto/KustoFunctions/KQLStringFunctions.cpp]: Changes are applied as suggested.

**Review by kashwy — APPROVED** [review]

File: src/Parsers/Kusto/KustoFunctions/KQLStringFunctions.cpp (hunk 1 above)
Comment (kashwy) [reply src/Parsers/Kusto/KustoFunctions/KQLStringFunctions.cpp]: @SadiHassan  merge it and let Larry know to test

Comment (SadiHassan) [reply src/Parsers/Kusto/KustoFunctions/KQLStringFunctions.cpp]: Merging as per @kashwy suggested. @larryluogit , you are good to test this change.

**Pull request merged**